### PR TITLE
`subclassed-final-class` documentation examples

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
|    | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
| ✓   | :scroll: Docs          |

## Description

Add the good & bad documentation examples.
Also add the message for the checker in the `check_messages` decorator.

Partially closes issue https://github.com/PyCQA/pylint/issues/5953

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
I guess  we want to keep the class final in the solution we proposed. So the fix is maybe to use an interface to be able to extend without inheriting. As all message related to design decision there's probably not one size fit all solution and we probably need to add the reasoning in details.rst.

File: pylint/checkers/classes/class_checker.py [hunk 1]
```diff
@@ -797,6 +797,7 @@ def _ignore_mixin(self):
         "duplicate-bases",
         "redefined-slots-in-subclass",
         "invalid-enum-extension",
+        "subclassed-final-class",
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/classes/class_checker.py:800]: I think we could add this change on main first so the CI is faster when we modify the documentation.

File: doc/data/messages/s/subclassed-final-class/good.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Base:
+    ...
+
+
+class MyClass(Base):
+    ...
```
[the remark this answers is not in the document]
Comment (mbyrnepr2) [reply doc/data/messages/s/subclassed-final-class/good.py:6]: > I guess we want to keep the class final in the solution we proposed. So the fix is maybe to use an interface to be able to extend without inheriting. As all message related to design decision there's probably not one size fit all solution and we probably need to add the reasoning in details.rst.

We could go this route of your suggested example. My thinking was that the examples would simply show:
- The message will be emitted when the class decorated with `@final` is sub-classed
- The message will not be emitted when the class decorated with `@final` is not sub-classed.

I could add this explanation to the details.rst to make that point clear as I'm sure the examples could be interpreted many ways, especially if you have never studied this checker.

[the remark this answers is not in the document]
Comment (mbyrnepr2) [reply doc/data/messages/s/subclassed-final-class/good.py:6]: And thank you for taking such a close look @Pierre-Sassoulas!

File: pylint/checkers/classes/class_checker.py (hunk 1 above)
Comment (mbyrnepr2) [reply pylint/checkers/classes/class_checker.py]: Completed in https://github.com/PyCQA/pylint/pull/6190.

File: doc/data/messages/s/subclassed-final-class/good.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbyrnepr2) [reply doc/data/messages/s/subclassed-final-class/good.py]: I've updated the examples treating the final class like an enum class for both examples.
If it isn't intuitive I can update to use your example - I thought this way looked a bit more straightforward but I could be wrong!

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**